go/worker/keymanager/watcher.go

```diff
@@ -313,41 +314,48 @@ func (w *rtNodeWatcher) watch(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-epoCh:
-			func() {
-				// Get executor committee members.
-				cms, err := w.consensus.Scheduler().GetCommittees(ctx, &scheduler.GetCommitteesRequest{
-					Height:    consensus.HeightLatest,
-					RuntimeID: w.runtimeID,
-				})
-				if err != nil {
-					w.logger.Error("failed to fetch runtime committee",
-						"err", err,
-					)
-					return
+			// Get executor committee members.
+			// Old ones need to be cleared out of the ACL even in case of errors.
+			clear(computeNodes)
+			cms, err := w.consensus.Scheduler().GetCommittees(ctx, &scheduler.GetCommitteesRequest{
+				Height:    consensus.HeightLatest,
+				RuntimeID: w.runtimeID,
+			})
+			if err != nil {
+				w.logger.Error("failed to fetch runtime committee",
+					"err", err,
+				)
+				break
+			}
+
+			// Get a list of all nodes at once, so we can match committee members with their descriptors.
+			allNodes, err := w.consensus.Registry().GetNodes(ctx, consensus.HeightLatest)
```

Comment:
Why are we fetching many nodes that we don't actually need? The previous version only fetches committee members.

Reply:
My comment was wrong, so this is actually my mistake. I thought we have method `GetNodesByID`, but we don't.